src/Aspire.Dashboard/Components/Pages/Resources.razor.cs

```diff
@@ -202,21 +206,28 @@ private bool ApplicationErrorCountsChanged(Dictionary<OtlpApplication, int> newA
         return false;
     }
 
-    private void ShowResourceDetails(ResourceViewModel resource)
+    private async Task ShowResourceDetailsAsync(ResourceViewModel resource, string buttonId)
     {
+        _elementIdBeforeDetailsViewOpened = buttonId;
+
         if (SelectedResource == resource)
         {
-            ClearSelectedResource();
+            await ClearSelectedResourceAsync();
         }
         else
         {
             SelectedResource = resource;
         }
     }
 
-    private void ClearSelectedResource()
+    private async Task ClearSelectedResourceAsync()
     {
         SelectedResource = null;
+
+        if (_elementIdBeforeDetailsViewOpened is not null)
+        {
+            await JS.InvokeVoidAsync("focusElement", _elementIdBeforeDetailsViewOpened);
+        }
     }
```

Comment:
You've changed this so the view button is always focused when the resource is closed, but there are a number of ways to close the resource. For example, if someone changes search parameters on a page then the resource is closed. We don't want to focus the button in that scenario.

What is the exact scenario where the view button should be focused?

Reply:
You can check the linked bug, but this is in reaction to the user manually closing the details panel.

> You've changed this so the view button is always focused when the resource is closed

Oops, thanks. Fixed